util/src/main/java/io/camunda/zeebe/util/sched/Actor.java

```diff
@@ -10,19 +10,38 @@
 import io.camunda.zeebe.util.CloseableSilently;
 import io.camunda.zeebe.util.Loggers;
 import io.camunda.zeebe.util.sched.future.ActorFuture;
+import java.util.HashMap;
+import java.util.Map;
 import java.util.concurrent.TimeUnit;
 import java.util.function.BiConsumer;
 import java.util.function.Consumer;
 
 public abstract class Actor implements CloseableSilently, AsyncClosable, ConcurrencyControl {
 
+  public static final String ACTOR_PROP_NAME = "actor-name";
+  public static final String ACTOR_PROP_PARTITION_ID = "partitionId";
+
   private static final int MAX_CLOSE_TIMEOUT = 300;
   protected final ActorControl actor = new ActorControl(this);
+  private Map<String, String> context;
 
   public String getName() {
     return getClass().getName();
   }
 
+  /**
+   * @return a map which defines the context where the actor is run. Per default it just returns a
+   *     map with the actor name. Ideally sub classes add more context, like the partition id etc.
+   */
+  public Map<String, String> getContext() {
```

Comment:
:question: Why not immediately instantiate the map on construction? Also, since any call to `getContext()` would mutate non thread safe state, I would prefer it. Generally the fact you can publicly access mutable state is probably a no-no since it's potentially sharing state in an unsafe way - but I get that `getContext` is meant to be called only from the actor itself (write) and the task (read). We could make the property protected (so the actor itself can write to `context`), and have this one here return an unmodifiable map. `Collections.unmodifiableMap` is just a view on the map which blocks writes, so you also wouldn't need to allocate it all the time.

I get that this may seem pedantic, but we should try to avoid sharing state with actors as much as possible even in cases where it seems safe, just to avoid this becoming more common place.

Reply:
With the context map it was a proposal, since I was not sure how otherwise to implement it for the actors.

Yes I expect that only the actor itself modifies it (and the task reads it, which is the same thread btw). Tbh. before I had the map creation inside the method, without having a field, which would be safer. But then I thought I create to many objects and added the lazy instantiation without thinking again about the thread safety sorry. You are right ofc. I haven't returned here a unmodifiable map since I expected to be overwritten by subclasses.